### ISSUE TITLE
algorithm on Bitnami ingress controller

### ISSUE THREAD

**Issue by rthamrin** [bitnami/bitnami-docker-nginx-ingress-controller#16]
### Name and Version

bitnami/bitnami/nginx-ingress-controller

### What is the problem this feature will solve?

1. I have installed the ingress controller with helm through helm list repository inside my Kubernetes Cluster, and I used nginx-ingress-controller-9.1.11. I read the default directory of bitnami is located in /opt/bitnami/nginx/conf/nginx.conf: but I did not find the file there but mine is /etc/nginx/nginx.conf , is something wrong with my installation. I use Ubuntu version 21.

2. what is the default algorithm of Nginx ingress? is it a round_robin algorithm? I try to exec my ingress controller pods. it shows EWMA . 
3. how to change the ingress controller algorithm?

### What is the feature you are proposing to solve the problem?

adding more algorithm

### What alternatives have you considered?

How to change the default algorithm to another algorithm? such as less_conn or ip_hash

**Comment by miguelaeh** [comment]
Hi @rthamrin ,
Could you share the exact command you used to install it, please?

About the algorithm, I am not fully sure right now, but it should appear in the [official docs](https://kubernetes.github.io/ingress-nginx/).

**Comment by rthamrin** [comment]
> Hi @rthamrin , Could you share the exact command you used to install it, please?
> 
> About the algorithm, I am not fully sure right now, but it should appear in the [official docs](https://kubernetes.github.io/ingress-nginx/).

just through helm

adding the repo
```
helm repo add bitnami https://charts.bitnami.com/bitnami
```
install the bitnami

```
helm install [bitnami-ingress] bitnami/nginx-ingress-controller -n [namespace]
```
I do some changes before installing the ingress

hostNetwork: True
Kind: DaemonSet
watchingresswithoutclass: true
dnspolicy: ClusterFirstWithHostNet
usehostPort: True
reportnodeinternal-ip-address=true"

**Comment by miguelaeh** [comment]
> where is actually my nginx.conf located, is it in etc/nginx.conf ?

 It is at `/opt/bitnami/nginx/conf/nginx.conf`.

Checking your logs the file seems to be at that location:

```
nginx: the configuration file /opt/bitnami/nginx/conf/nginx.conf syntax is ok
nginx: configuration file /opt/bitnami/nginx/conf/nginx.conf test is successful
```

**Comment by miguelaeh** [comment]
Hi @rthamrin ,

The installed by our chart is https://kubernetes.github.io/ingress-nginx/

And the `nginx.conf` should be there. Are you using the bitnami image?

Running the following I can verify the file is there:

```
docker run --rm -it --entrypoint bash bitnami/nginx-ingress-controller
I have no name!@119a1541be15:/opt/bitnami/nginx-ingress-controller$ ls /opt/bitnami/nginx/conf/nginx.conf
/opt/bitnami/nginx/conf/nginx.conf
```

**Comment by rthamrin** [comment]
> Hi @rthamrin ,
> 
> The installed by our chart is https://kubernetes.github.io/ingress-nginx/
> 
> And the `nginx.conf` should be there. Are you using the bitnami image?
> 
> Running the following I can verify the file is there:
> 
> ```
> docker run --rm -it --entrypoint bash bitnami/nginx-ingress-controller
> I have no name!@119a1541be15:/opt/bitnami/nginx-ingress-controller$ ls /opt/bitnami/nginx/conf/nginx.conf
> /opt/bitnami/nginx/conf/nginx.conf
> ```

I installed the ingress through helm without docker.

```
helm install [bitnami-ingress] bitnami/nginx-ingress-controller -n [namespace]
```

**Comment by miguelaeh** [comment]
Yes, but the commands above show that the file must be there, it is not needed to add the file on your own

**Comment by rthamrin** [comment]
> Yes, but the commands above show that the file must be there, it is not needed to add the file on your own

so what do you suggest me to do?

**Comment by miguelaeh** [comment]
did you override the image when installing the chart?

**Comment by rthamrin** [comment]
> did you override the image when installing the chart?

I am not sure, but I run YAML file on my cluster . and I have /etc/nginx/nginx.conf , but not sure where and when the file was created. but as I remember install the nginx-ingress-controller-9.1.11 does create any nginx.conf or folder /opt/bitnami
```
apiVersion: apps/v1
kind: Deployment 
metadata:
  name: nginx-deployment
  namespace: ingress
spec:
  replicas: 5
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx-deployment
        image: nginx:1.20.0
        ports:
        - containerPort: 80
````

**Comment by miguelaeh** [comment]
Checking that file, you are not using the bitnami image, so it has sense that the file does not exist. Change the image to the bitnami one, or change the chart to follow that image files structure.

**Comment by rthamrin** [comment]
> Checking that file, you are not using the bitnami image, so it has sense that the file does not exist. Change the image to the bitnami one, or change the chart to follow that image files structure.

Thank you for your input, I just realized that image, but unfortunately, the folder is still not created.
1. my machine is K3S Kubernetes
2. I delete the old deployment and service
3. I change the image to bitnami/nginx:latest from https://hub.docker.com/r/bitnami/nginx
4. I created a new  ```kubectl create -f file.yaml```
5. deployment is created
6. expose the deployment to service use port 80
7. pods are running
8. cd /opt/
9. checking bitnami folder, but still not there.

**Comment by miguelaeh** [comment]
Are you using persistence? Do you have some PVC from the older deployment? In that case, try to delete them and re-deploy, so the image is able to re-initialize

**Comment by rthamrin** [comment]
> Are you using persistence? Do you have some PVC from the older deployment? In that case, try to delete them and re-deploy, so the image is able to re-initialize

will you please elaborate on how to check "persistence" and PVC.

me also found the fact .

```# ps auxf | grep nginx
root      368000  0.0  0.0  11384  2336 pts/1    S+   07:36   0:00  |   |               \_ grep --color=auto nginx
master      3097  0.1  0.6 743824 27608 ?        Ssl  Apr04   2:48  \_ nginx-ingress-controller /nginx-ingress-controller --default-backend-service=ingress/ingress-controller-nginx-ingress-controller-default-backend --election-id=ingress-controller-leader --controller-class=k8s.io/ingress-nginx --configmap=ingress/ingress-controller-nginx-ingress-controller --report-node-internal-ip-address=true --watch-ingress-without-class=true

master      8212  0.0  0.0 158344  1160 ?        S    Apr04   0:00      \_ nginx: master process /usr/local/nginx/sbin/nginx -c /etc/nginx/nginx.conf

master      8253  0.0  0.1 429492  5944 ?        Sl   Apr04   1:28          \_ nginx: worker process

master      8254  0.0  0.0 155992   472 ?        S    Apr04   0:00          \_ nginx: cache manager process

master      6642  0.0  0.0  15212   616 ?        Ss   Apr04   0:00  \_ nginx: master process /opt/bitnami/nginx/sbin/nginx -c /opt/bitnami/nginx/conf/nginx.conf -g daemon off;
master      7071  0.0  0.0  15640   508 ?        S    Apr04   0:07      \_ nginx: worker process

root        7403  0.0  0.0   8868    72 ?        Ss   Apr04   0:00  \_ nginx: master process nginx -g daemon off;

systemd+    7721  0.0  0.0   9252   464 ?        S    Apr04   0:03      \_ nginx: worker process

root        7924  0.0  0.0   6288   224 ?        Ss   Apr04   0:00  \_ nginx: master process nginx -g daemon off;

systemd+    7978  0.0  0.0   6760   940 ?        S    Apr04   0:08      \_ nginx: worker process

root      144255  0.0  0.0   6028   212 ?        Ss   Apr05   0:00  \_ nginx: master process nginx -g daemon off;

systemd+  144369  0.0  0.0   6492   784 ?        S    Apr05   0:01      \_ nginx: worker process
```

**Comment by miguelaeh** [comment]
I suggest you to clean the whole environment and re-deploy using the correct images so they are able to re-configure properly.

**Comment by rthamrin** [comment]
> I suggest you to clean the whole environment and re-deploy using the correct images so they are able to re-configure properly.

no other way? do you mean install my Kubernetes from the beginning? it seems impossible to do

**Comment by miguelaeh** [comment]
Just remove PVCs if you don't care about the stored data
